nixos/modules/services/mail/postsrsd.nix

```diff
@@ -7,16 +7,52 @@
 let
 
   cfg = config.services.postsrsd;
+  runtimeDirectoryName = "postsrsd";
+  runtimeDirectory = "/run/${runtimeDirectoryName}";
+  # TODO: follow RFC 42, but we need a libconfuse format first:
+  #       https://github.com/NixOS/nixpkgs/issues/401565
+  # Arrays in `libconfuse` look like this: {"Life", "Universe", "Everything"}
+  # See https://www.nongnu.org/confuse/tutorial-html/ar01s03.html.
+  #
+  # Note: We're using `builtins.toJSON` to escape strings, but JSON strings
+  # don't have exactly the same semantics as libconfuse strings. For example,
+  # "${F}" gets treated as an env var reference, see above issue for details.
+  libconfuseDomains = "{ " + lib.concatMapStringsSep ", " builtins.toJSON cfg.domains + " }";
+  configFile = pkgs.writeText "postsrsd.conf" ''
+    secrets-file = "''${CREDENTIALS_DIRECTORY}/secrets-file"
+    domains = ${libconfuseDomains}
+    separator = "${cfg.separator}"
+    socketmap = "unix:${runtimeDirectory}/socket"
+
+    # Disable postsrsd's jailing in favor of confinement with systemd.
+    unprivileged-user = ""
+    chroot-dir = ""
+  '';
```

Comment:
@bennofs this is where the new 2.0 config file is written. There is ongoing work in #401565 to support arbitrary options via a freeform `settings` option, but currently the module only supports a few explicitly maintained options.

> - ``srs-domain`` to the value from ``SRS_DOMAIN``

This means the old (removed) `domain` option should map to a config file option `srs-domain`. @jfly was this just missed during the PR? Could we add the removed `domain` option back and use it to set `srs-domain` if the option is defined?

> - ``domains`` to the list of values from ``SRS_EXCLUDE_DOMAINS``

This means if you were configuring `excludeDomains` in 1.0, you should now configure `domains` in 2.0.

We _do_ have a `domains` option, so this should be possible.

Reply:
Urg, sorry @bennofs, this was more confusing than I realized.

You can just set the new `domains` option to whatever you were setting `domain` to before. You don't have to set `srs-domain` explicitly, see https://github.com/roehling/postsrsd/blob/2.0.10/doc/postsrsd.conf#L32C19-L32C71:

> If unset, the first configured local domain is used.

> @jfly is it therefore possible for us to replace the excludeDomains removal module with a rename module?

Yes, that would be correct. I intentionally didn't do this because I wanted to draw people's attention to the v2 release notes. Probably not the right way to do this.